src/main/java/com/programmers/vouchermanagement/repository/voucher/MemoryVoucherRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.programmers.vouchermanagement.repository.voucher;
+
+import com.programmers.vouchermanagement.domain.voucher.Voucher;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+@Profile("local")
+public class MemoryVoucherRepository implements VoucherRepository {
+    private final Map<UUID, Voucher> storage = new HashMap<>();
+
+    public void save(Voucher voucher) {
+        storage.put(voucher.getId(), voucher);
+    }
+
+    public List<Voucher> findAll() {
+        return new ArrayList<>(storage.values());
```

Comment:
ArrayList가 아닌 stream()의 toList를 사용하면 불변 리스트가 되어 조회할 때 좀 더 안전하게 조회할 수 있습니다. (조회 후, 리스트 수정 방지)
```suggestion
        return storage.values()
        .stream()
        .toList();
```